pkg/cache/cluster.go

```diff
@@ -782,6 +782,10 @@ func (c *clusterCache) IterateHierarchy(key kube.ResourceKey, action func(resour
 	defer c.lock.RUnlock()
 	if res, ok := c.resources[key]; ok {
 		nsNodes := c.nsIndex[key.Namespace]
+		if key.Namespace == "" {
```

Comment:
This will affect controller performance. The `IterateHierarchy` method is already the main CPU consumer. I think the simplest fix could be adding `childredByParent map[kube.ResourceKey][]kube.ResourceKey` index: the map that holds parent ref and list of child refs. The map should be updated on the fly in `onNodeUpdated` and `onNodeRemoved` methods.

Reply:
Hi @alexmt! Sorry for delaying this. I have addressed your comment. Please take a look.

Reply:
is this ready now?